src/test/java/com/fasterxml/jackson/databind/deser/enums/EnumDeserializer5271Test.java

```diff
@@ -0,0 +1,43 @@
+package com.fasterxml.jackson.databind.deser.enums;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class EnumDeserializer5271Test
+{
+    public enum MyEnum {
+        T10("10%"), T20("20%"), T30("30%");
+
+        private final String code;
+
+        MyEnum(String code) {
+            this.code = code;
+        }
+
+        @JsonValue
+        public String getCode() {
+            return code;
+        }
+    }
+
+    @Test
+    void convertStringToEnum() {
+        _testConvert(JsonMapper.builder().disable(DeserializationFeature.READ_ENUMS_USING_TO_STRING)
+                        .build()
+        );
```

Comment:
This one would pass since `DeserializationFeature.READ_ENUMS_USING_TO_STRING` is default behavior.